packages/headless/src/controllers/core/smart-snippet-questions-list/headless-core-smart-snippet-interactive-questions.ts

```diff
@@ -40,8 +43,9 @@ export interface SmartSnippetInteractiveQuestions {
 /**
  * @internal
  */
-export function buildSmartSnippetInteractiveQuestions(
-  engine: SearchEngine,
+export function buildCoreSmartSnippetInteractiveQuestions(
+  engine: CoreEngine,
+  analyticsClient: SmartSnippetAnalyticsClient,
```

Comment:
Instead of adding the analytics client as an option here, I think we should copy the signature of the core interactive result controller and have the third parameter be `action: () => void`.

I think it's fine that `props` would become required.

Usage example:
```ts
buildCoreSmartSnippetInteractiveQuestions(
  engine,
  {
    selectionDelay: props?.options?.selectionDelay,
  },
  ({source, questionAnswerId}) => {
    engine.dispatch(
      analyticsClient.logOpenSmartSnippetSuggestionSource({
        questionAnswerId,
      })
    );
    engine.dispatch(pushRecentResult(source));
  }
);
```

Reply:
To remain consistant with the already existing pattern I completely removed the `headless-core-smart-snippet-interactive-questions` file.
Now we have two separate files one specific to the search use case and another one specific to the insight use case.